beacon-chain/state/state-native/setters_misc.go

```diff
@@ -114,6 +114,10 @@ func (b *BeaconState) SaveValidatorIndices() {
 		return
 	}
 
+	if b.validatorIndexCache == nil {
```

Comment:
does initializing the cache here make sense?

Reply:
Shouldn't that be during the lock?

Reply:
hmm that's true

Reply:
accepted suggestion *